### ISSUE TITLE
Unpack/Pack issue between v9 and c8.2

### ISSUE THREAD

**Issue by hmdvs** [scottdurow/SparkleXrm#249]
Hi Scott

No problems unpack/pack a solution with following.
-spkl 1.0.198
-Microsoft.CrmSdk.CoreTools 8.2.05

Solution imports with no issue into crm online Version 1612 (9.0.2.449) (DB 9.0.2.449)

With the above I am unable to deploy plugins or generate earlybound entities and get following error:

>>
CrmSvcUtil Error: 2 : Exiting program with exit code 2 due to exception : System.IO.FileLoadException: Could not load file or assembly 'CrmSvcUtil, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
Exiting program with exception: Could not load file or assembly 'CrmSvcUtil, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'CrmSvcUtil, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMarkHandle stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName, ObjectHandleOnStack type)

   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName)

----

The problem is that my plugins reference CoreAssemblies 9.0.0.7. When I update Microsoft.CrmSdk.CoreTools to 9.0 then plugins and earlybound entities run with no issues but then cause my solution import fails with invalid xml on Visualizations.

Same when install spkl in the plugins project directly as it picks up the first CoreTools in my packages
spkl.1.0.198\tools\..\..\Microsoft.CrmSdk.CoreTools.8.2.0.5\content\bin\coretools\CrmSvcUtil.exe

**Comment by scottdurow** [comment]
Can you try the latest build- https://www.nuget.org/packages/spkl/1.0.225-beta

It’s going to be the next release version shortly.

**Comment by hmdvs** [comment]
with the new version and CoreTools 9.0.0.7 I am able to deploy plugins and create earlybound but unable to import the packed solution, error:

_The element 'Visualizations' has invalid child element 'savedqueries'. List of possible elements expected: 'visualization'._

If you have multiple spkl references, ie. one for the solution and other for plugins, it still picks up the first within packages as below:
_spkl.1.0.198\tools\..\..\Microsoft.CrmSdk.CoreTools.8.2.0.5\_


So basically with 1.0.198 with CoreTools 8.2.0.5 I can successfully unpack/pack and import solutions - Solution project is separate from Plugins project.

Plugins deploy correctly with spkl.1.0.198 and CoreTools.9.0.0.7 - but Earlybound fails.

I think for now I will generate Earlybound entities seperate.

It looks like unpack/pack with CoreTools 9+ generates different XML schema than 8.2 hence the Solution import failure.

**Comment by scottdurow** [comment]
@hmdvs Is this still an issue for you with the least release?

**Comment by hmdvs** [comment]
@scottdurow  I've not re looked at this issue since. Will give it a test on the new version and report results.

**Comment by hmdvs** [comment]
@scottdurow - I re tested unpack/pack and it looks good with the following:

crm:Version 1612 (9.0.2.751) 
spkl:1.0.226
CoreTools: 9.0.0.7

**Comment by mihirkadam** [comment]
I tried the new one and not facing any issue :)
